src/api/tests/api/v1/routers/test_statique.py

```diff
@@ -833,3 +887,56 @@ def test_bulk_update(client_auth, db_session):
         .one()
         .paiement_cb
     )
+
+
+def test_bulk_update_audits(client_auth, db_session):
+    """Test that bulk endpoint updates audits."""
+    size = 10
+    statiques = StatiqueFactory.batch(
+        size=size,
+        paiement_cb=False,
+    )
+    save_statiques(db_session, statiques)
+
+    # Update paiement_cb field
+    statiques[3].paiement_cb = statiques[7].paiement_cb = True
+
+    payload = [json.loads(s.model_dump_json()) for s in statiques]
+    response = client_auth.post("/statique/bulk", json=payload)
+    assert response.status_code == status.HTTP_201_CREATED
+
+    # Get user requesting the server
+    user = db_session.exec(select(User).where(User.email == "john@doe.com")).one()
+
+    # Check changes and audit
+    pdc3 = db_session.exec(
+        select(PointDeCharge).where(
+            PointDeCharge.id_pdc_itinerance == statiques[3].id_pdc_itinerance
+        )
+    ).one()
+    assert pdc3.paiement_cb
+    assert len(pdc3.audits) == 1
+    assert pdc3.audits[0].author_id == user.id
+
+    pdc7 = db_session.exec(
+        select(PointDeCharge).where(
+            PointDeCharge.id_pdc_itinerance == statiques[7].id_pdc_itinerance
+        )
+    ).one()
+    assert pdc7.paiement_cb
+    assert len(pdc7.audits) == 1
```

Comment:
Maybe it fails because when listening the "after_update" event in an upsert context, the event is never raised on "conflict do update"?

Reply:
Maybe we should create a custom event?